### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by RoboRambo**
Not sure if needed/wanted. Maybe should be left for when we go crates.io.

Result:
image
image

### CODE REVIEW

**Comment by enthusiastmartin** [comment]
Yes sure, 

within this PR, please add also description for the remaining functions after merging master and we're good to go with v1.0.0

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ authors = ['GalacticCouncil']
 edition = "2018"
 
 [dependencies]
-primitive-types = {default-features = false, version = '0.7.2'}
+primitive-types = {default-features = false, version = '0.9.0'}
```
Comment (enthusiastmartin) [inline Cargo.toml]: why changing the version in this PR when we only updating comments/docs?

File: Cargo.lock [hunk 2]
```diff
@@ -16,13 +16,19 @@ checksum = "7a81dae078cea95a014a339291cec439d2f232ebe854a9d672b796c6afafa9b7"
 
 [[package]]
 name = "fixed-hash"
-version = "0.6.1"
+version = "0.7.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "11498d382790b7a8f2fd211780bec78619bba81cdad3a283997c0c41f836759c"
+checksum = "cfcf0ed7fe52a17a03854ec54a9f76d6d84508d1c0e66bc1793301c73fc8493c"
```
Comment (enthusiastmartin) [inline Cargo.lock]: Why lock file changes?  it is due to the version change in toml file, but why needed in this PR?

File: src/amm.rs [hunk 3]
```diff
@@ -110,6 +110,14 @@ pub fn calculate_in_given_out(out_reserve: Balance, in_reserve: Balance, amount_
     round_up!(result.unwrap())
 }
 
+/// Calculating required amount of asset b given asset a.
+/// Formula : AMOUNT * ASSET_B_RESERVE / ASSET_A_RESERVE
+///
+/// - `asset_a_reserve` - reserve amount of asset a
+/// - `asset_b_reserve` - reserve amount of asset b
+/// - `amount` - buy amount
```
Comment (enthusiastmartin) [inline src/amm.rs]: Not buy amount -> liquidity amount

File: src/amm.rs [hunk 4]
```diff
@@ -122,6 +130,15 @@ pub fn calculate_liquidity_in(asset_a_reserve: Balance, asset_b_reserve: Balance
     to_balance!(b_required_hp)
 }
 
+/// Calculating amount of assets returned when removing liquidity.
+/// Formula A: AMOUNT * ASSET_A_RESERVE / TOTAL_LIQUIDITY
+/// Formula B: AMOUNT * ASSET_B_RESERVE / TOTAL_LIQUIDITY
+///
+/// - `asset_a_reserve` - reserve amount of asset a
+/// - `asset_b_reserve` - reserve amount of asset b
+/// - `amount` - buy amount
```
Comment (enthusiastmartin) [inline src/amm.rs]: same here

File: src/lib.rs [hunk 5]
```diff
@@ -1,6 +1,14 @@
+//! # HydraDX Math
+//!
+//! A collection of utilities to make performing liquidity pool
+//! calculations more convenient.
+
 #![cfg_attr(not(feature = "std"), no_std)]
 
+/// !! Insert Description here. !!
```
Comment (enthusiastmartin) [inline src/lib.rs]: So lets insert something here.

Comment (RoboRambo) [reply src/lib.rs]: Not so sure what to insert, i'd rather not insert
"A collection of utilities..." a third time :D

Comment (enthusiastmartin) [reply src/lib.rs]: so just delete it.

**Review by enthusiastmartin — APPROVED** [review]

**Pull request merged**